src/components/DropdownItem.js

```diff
@@ -0,0 +1,374 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { CheckIcon } from './icons';
+
+import defaultTheme from '../themes/DefaultTheme';
+import darkTheme from '../themes/DarkTheme';
+
+const DropdownItem = ({
+  title,
+  description,
+  Icon,
+  className,
+  active,
+  onClick,
+  href,
+  target,
+  children,
+  disabled,
+  noItemsActiveState,
+}) => {
+  const titleText = typeof title === 'string' ? title : title.text;
+  const ActionElem = href ? 'a' : 'button';
+
+  return (
+    <div
+      className={classnames(
+        'drui-dropdownItem',
+        {
+          [className]: className,
+          ['drui-dropdownItem--withDescription']: description,
+          ['drui-dropdownItem--active']: active,
+          ['drui-dropdownItem--disabled']: disabled,
+          ['drui-dropdownItem--noItemsActiveState']: noItemsActiveState,
+        })
+      }
+      onClick={onClick}
+      key={`drui-dropdown-item-${titleText.split(' ').join('').toLowerCase()}`}
+    >
+      <ActionElem
+        href={href || null}
+        type={!href ? 'button' : null}
+        className="drui-dropdownItem__action"
+        target={target || null}
+        rel={target === '_blank' ? 'noopener nofollow' : null}
+        disabled={ActionElem === 'button' ? disabled : null}
+      >
+        <div className="drui-dropdownItem__iconWrapper">
+          {/* Custom icon for dropdown item */}
+          {Icon &&
+            <Icon className="drui-dropdownItem__icon" />
+          }
+
+          {/* No custom icon, menu item is active */}
+          {!noItemsActiveState && !Icon && active &&
+            <CheckIcon className="drui-dropdownItem__icon" />
+          }
+        </div>
+
+        {/* Item title */}
+        <div className="drui-dropdownItem__title">
+          <span className="drui-dropdownItem__titleText">{titleText}</span>
+
+          {typeof title === 'object' && title.Icon &&
+            <title.Icon className="drui-dropdownItem__titleIcon" />
+          }
+        </div>
+
+        {/* Item description */}
+        {description &&
+          <span className="drui-dropdownItem__description">{description}</span>
+        }
+      </ActionElem>
+
+      {children || null}
```

Comment:
I couldn't find any example of using this in stories and I can't come up with one now, so I would suggest to remove the option of passing children to that component.